src/libcore/array/mod.rs

```diff
@@ -202,6 +204,65 @@ where
     }
 }
 
+#[stable(feature = "array_from_iter_impl", since = "1.44.0")]
+impl<T, const N: usize> FromIterator<T> for [T; N] 
+where
+     [T; N]: LengthAtMost32,
+{
+    #[inline]
+    fn from_iter<I: IntoIterator<Item = T>>(iter: I) -> Self {
+        <Self as ArrayFromIter<T, I::IntoIter>>::from_iter(iter.into_iter())
+    }
+}
+
+// Specialization trait used for array::from_iter
+trait ArrayFromIter<T, I> {
+    fn from_iter(iter: I) -> Self;
+}
+
+impl<T, I, const N: usize> ArrayFromIter<T, I> for [T; N] 
+    where I: Iterator<Item=T>,
+    [T; N]: LengthAtMost32,
+{
+    default fn from_iter(mut iter: I) -> Self {
```

Comment:
[The language team has imposed a moratorium on new uses of specialization at this time.](https://forge.rust-lang.org/libs/maintaining-std.html#is-specialization-involved) Please remove the specializations for now.